### DISCUSSION TITLE
Something confused about link-citations?

### DISCUSSION THREAD

**Discussion by Wattskemov** [jgm/pandoc#9729] [Q&A]
I use _link-citations: true_ to generate Hugo markdown( or you can say markdown_strict), the command I am using is:

```
pandoc --citeproc --bibliography reference.bib --csl ieee.csl -t markdown_strict -o result.md index.md
```

I am using "link-citations: true" in my index.md file to generate citation links. There is a strange thing here, the reference list generated by pandoc doesn't contain any ID property to let linking. Here is the result:

```
<span class="csl-left-margin">\[1\]
</span><span class="csl-right-inline">“VikParuchuri/marker: Convert PDF
to markdown quickly with high accuracy.”
<https://github.com/VikParuchuri/marker>, 2024.</span>

<span class="csl-left-margin">\[2\]
</span><span class="csl-right-inline">“Top 5 google scholar APIs and
scrapers in 2024.”
<https://blog.apify.com/best-google-scholar-apis-scrapers/>,
2024.</span>
```

If I generate HTML result of the above markdown file, the result will contain ID property:

```
<div id="ref-VikParuchuri-marker" class="csl-entry" role="listitem">
<div class="csl-left-margin">[1] </div><div class="csl-right-inline"><span>“VikParuchuri/marker: Convert PDF to
markdown quickly with high accuracy.”</span> <a href="https://github.com/VikParuchuri/marker" class="uri">https://github.com/VikParuchuri/marker</a>, 2024.</div>
</div>
<div id="ref-Top-2024-05-06" class="csl-entry" role="listitem">
<div class="csl-left-margin">[2] </div><div class="csl-right-inline"><span>“Top 5 google scholar APIs and scrapers
in 2024.”</span> <a href="https://blog.apify.com/best-google-scholar-apis-scrapers/" class="uri">https://blog.apify.com/best-google-scholar-apis-scrapers/</a>,
2024.</div>
</div>
```

So I'm curious, in the markdown output, if there is no ID attribute, in the final HTML page rendered by Hugo, there is no way for the citation of a reference to jump to the corresponding entry in the Reference list. If I want to make sure that a reference citation jumps to the corresponding reference entry, I need to manually put in an ID similar to the HTML output above.

So, is there any way to write this ID into the span tag when using the link-citations parameter to convert the markdown file?

Also, I'm wondering, is this considered a bug?

**Comment by jgm** [comment]
Seems like a bug, if the markdown writer is generating links but the anchors aren't there.  Please submit an issue for this.

**Reply by Wattskemov** [reply]
OK. An issue has been created. Thanks for checking this.
